### PR TITLE
[1] Dna exten/itp

### PR DESCRIPTION
**Pull request by fgrunewald**
To Do

- [x] add tests for errors in gen_dna
- [x] add tests for missing edge warning
- [x] fix ig parsing 1 line

### CODE REVIEW

**Comment by fgrunewald** [comment]
@jan-stevens whenever you've got time for a review ;)

File: polyply/src/simple_seq_parsers.py
Comment (jan-stevens) [inline polyply/src/simple_seq_parsers.py:259]: Dropping the first item in the list 'clean_lines' breaks the ig reader (tested it on the Single stranded circular DNA tutorial).

```suggestion
    seq_graph = _parse_plain(clean_lines, DNA=DNA, RNA=RNA, AA=AA)
```

Comment (jan-stevens) [reply polyply/src/simple_seq_parsers.py:259]: Just realized that in the ig file format, there should also be a sequence name between the comments and the sequence. So this is probably an error in the tutorial and not here?

File: polyply/src/meta_molecule.py
Comment (jan-stevens) [inline polyply/src/meta_molecule.py:125]:
```suggestion
        self.max_resid += 1
```

File: polyply/src/gen_dna.py
Comment (jan-stevens) [inline polyply/src/gen_dna.py:92]:
```suggestion
            meta_molecule.add_monomer(new_node, resname, [(correspondance[prev_node], total+1)])
```

**Review by jan-stevens — CHANGES_REQUESTED** [review]
Looks good, it will be very useful!
Only added some small comments.

File: polyply/src/simple_seq_parsers.py
Comment (fgrunewald) [reply polyply/src/simple_seq_parsers.py:259]: Yeah, I made this mistake a couple of times as well. The warning is literally issued a few lines before. Did it not print the warning? I also checked the online tutorial and to me it looks fine. From where did you copy the sequence?

Comment (jan-stevens) [reply polyply/src/simple_seq_parsers.py:259]: I copied the sequence from [here](https://github.com/marrink-lab/polyply_1.0/wiki/Tutorial:-Single-stranded-circular-DNA), where the comment does not specify that it is 'DNA'. 

I previously commented out the sequence name to troubleshoot and forgot I modified it. Adding 'DNA' to the comment line fixes the problems. Now I also realize that the warning a few lines above is actually for the sequence name... It makes sense now!

Comment (fgrunewald) [reply polyply/src/simple_seq_parsers.py:259]: @jan-stevens then it's resolved and ready to merge after your approval?

Comment (jan-stevens) [reply polyply/src/simple_seq_parsers.py:259]: Yes!

**Review by jan-stevens — APPROVED** [review]

**Pull request merged**
